src/main/webapp/script.js

```diff
@@ -82,12 +84,24 @@ function drawRegionsMap(censusDataArray) {
 // createDataArray takes in the data array returned by the census API
 // and reformats it into a data table for the visualization API.
 function createDataArray(censusDataArray) {
-  const vizDataArray = [];
-  censusDataArray.forEach((state) => {
-    vizDataArray.push([state[0], state[1]/1000]);
-  });
+  const vizDataArray = []
+  if (censusDataArray[0].length == 4) {
+    censusDataArray.forEach((state) => {
+      vizDataArray.push([state[0], percentToTotal(state[1], state[2])]);
+    });
+  } else {
+    censusDataArray.forEach((state) => {
+      vizDataArray.push([state[0], state[1]/1000]);
+    });
+  }
   // Changes the header of the vizDataArray to match the Visualization API
   vizDataArray[0][0] = 'State';
   vizDataArray[0][1] = 'Population';
```

Comment:
It's not always population anymore so we will need to change this (maybe just hardcode by using a small dictionary, and having its keys be the possible `action` strings from the form?)

Reply:
Done! Not very grammatically correct but should do the trick for now